### PR TITLE
Scroll optimize

### PR DESCRIPTION
**Pull request by singularity-s0**
Still need to:

- [x] fix scroll controller

### CODE REVIEW

File: lib/page/platform_subpage.dart [hunk 1]
```diff
@@ -167,6 +167,7 @@ abstract class PlatformSubpageState<T extends PlatformSubpage>
     return PrimaryScrollController(
       controller: _buildPrimaryScrollController(context),
       child: PlatformScaffold(
+          backgroundColor: Theme.of(context).scaffoldBackgroundColor,
```
Comment (w568w) [inline lib/page/platform_subpage.dart:170]: What is this line for? Does the `PlatformScaffold` miss a background?

Comment (singularity-s0) [reply lib/page/platform_subpage.dart:170]: Yes, on iOS.

File: lib/page/home_page.dart [hunk 2]
```diff
@@ -375,7 +375,7 @@ class HomePageState extends State<HomePage> with WidgetsBindingObserver {
     if (_needSendToWatch &&
         _preferences!.containsKey(SettingsProvider.KEY_FDUHOLE_TOKEN)) {
```
Comment (w568w) [inline lib/page/home_page.dart:376]: Can we eliminate `_preferences` which breaks the uniqueness of `SharedPreferences` held by `SettingsProvider`?

File: lib/page/platform_subpage.dart (hunk 1 above)
Comment (w568w) [reply lib/page/platform_subpage.dart:170]: > Yes, on iOS.

Then is it enough to add the patch *only* here? Has the line covers all `PlatformScaffold`s?

File: lib/page/home_page.dart (hunk 2 above)
Comment (singularity-s0) [reply lib/page/home_page.dart:376]: Yes. Can change this line to use `SettingsProvider`

File: lib/page/platform_subpage.dart (hunk 1 above)
Comment (singularity-s0) [reply lib/page/platform_subpage.dart:170]: I was expecting that `Scaffold`s would automatically use `scaffoldBackgroundColor` from `Theme`, but for some reason it wasn't. This is some kind of workaround solution.

**Comment by w568w** [comment]
Until we figure out a fix to all broken features (i.e. ScrollBar cannot scroll, double tap on title cannot jump to the top), this PR should not be merged.

**Comment by w568w** [comment]
Also PTAL at issue #95. If we uses a paging method like [TieBa](https://tieba.baidu.com/index.html), there should be pretty less problems and bugs.

**Comment by w568w** [comment]
There are some problems preventing this merge:

- [ ] On iOS, after using scroll-to-bottom in `hole_detail`, double tapping on the notification bar (which triggers a scroll in `PrimaryScrollController`) weirdly makes the list scroll to the bottom again, rather than to the top.
- [ ] If entering `hole_detail` with a floor located (e.g. from search results or other's references), the `ScrollBar` on the page looks good but cannot be dragged, throwing an exception indicating that there is no `ScrollPosition` attached to the `PrimaryScrollController` controlled by the scroll bar.
- `scrollable_positioned_list`'s bug: if item's size changes during scrolling by index, list will scroll to an incorrect position. See https://github.com/google/flutter.widgets/issues/330. It seems that the repository is out of maintenance, so we won't expect an official fix for the issue in the short term. (It is NOT a big problem if we drop usage of [keframe](https://github.com/w568w/keframe), because our item widgets' height will not change without it.)

However, removing `keframe` will bring pros and cons. It does have some issues, like slowing down the normal loading speed even under low render pressure, i.e. user is scrolling the list slowly,  but still seeing the following items' transformation from blank to a normal widget. But it really optimizes the frame lag caused by rapid scrolling, giving users a better experience when scrolling.

What do you think towards these problems? @singularity-s0

**Comment by singularity-s0** [comment]
Is it possible to use `Scrollable.ensureVisible()` for locating items instead?

**Comment by w568w** [comment]
> Is it possible to use `Scrollable.ensureVisible()` for locating items instead?

Looks hopeful. You can start a new branch and do some tests on it.

**Comment by singularity-s0** [comment]
> > Is it possible to use `Scrollable.ensureVisible()` for locating items instead?
> 
> Looks hopeful. You can start a new branch and do some tests on it.

Tests complete. You may revert to the previous implementation and merge these branches.

**Comment by w568w** [comment]
The PR and the branch is verified as unworkable. Close it.

**Comment by w568w** [comment]
Can we have a try on <https://github.com/superlistapp/super_sliver_list>? It looks promising.

**Pull request closed without merging**